src/tree/RefreshManager.ts

```diff
@@ -156,7 +163,13 @@ export class RefreshManager extends vscode.Disposable {
 
                         throw error;
                     }
+                } finally {
+                    // Move the next start timestamp to the current end timestamp
+                    eventsSinceTimestamp = eventsUntilTimestamp;
                 }
+
+                // If the event generator terminates on its expected lifecycle it will exit without an error
+                // Continue the loop as normal, starting the next event stream at the previous stop time

```

Comment:
Very much like how you fixed this with such a small change.

Should we reset `errorCount` here?

Reply:
That's a good thought. I'm still inclined toward "no", we should not reset it. Here's my thinking--

1. This is a "best-effort" sort of feature, it isn't essential to the extension or explorer working
2. The `errorCount` thing is basically to limit the maximum number of tries we make. Though I'm not sure how it could actually happen, say we're in some scenario where every other call fails. If we reset the `errorCount`, we'll keep doing this indefinitely; if we don't reset it, we'll encounter at most 3 failures before giving up.

Reply:
Sounds reasonable. Maybe add 1-line comment that giving up after max tries is the intention, so no resetting of error count.